notebooks/query-language-object-based-dialect/01.01.quantifier_capabilities.ipynb

```diff
@@ -0,0 +1,842 @@
+    "\n",
+    "This is an interesting profile because it covers a relatively large number of nodes and also spends considerable time in MPI communication routines."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "04ab3f46",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import hatchet as ht\n",
+    "gf = ht.GraphFrame.from_caliper(\"../../data/lulesh-16nodes/lulesh-annotation-profile-512cores.json\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6f37a13b-453e-4107-ac1c-b6025e390541",
+   "metadata": {},
+   "source": [
+    "### Displaying a Hatchet GraphFrame\n",
+    "A compact overview of a hatchet GraphFrame can be obtained using the `gf.tree()` function. We use this throughout the notebook to display the differences between an original GraphFrame and the resulting GraphFrame after applying a query."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4d29e959",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(gf.tree())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "152c81ff-4c15-4b5a-a7ba-6cb2c44d7479",
+   "metadata": {},
+   "source": [
+    "### Displaying a DataFrame\n",
+    "An additional detail perspective can be obtained by viewing the underlying data using a **DataFrame**. Hatchet uses pandas **DataFrames** to store the data on each node of the hierarchy and keeps the graph relationships between the nodes in a different data structure that is kept consistent with the **DataFrame**. The **DataFrame** holds all the numerical and categorical data associated with each node. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3bc1a4c7",
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "gf.dataframe"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f8f3341c-afc8-43c9-915a-ec6bc250cbf3",
+   "metadata": {},
+   "source": [
+    "### Dropping index levels\n",
+    "\n",
+    "As a precursor to defining queries, we drop the index level of the GraphFrame using the `drop_index_levels()` Hatchet function. Hatchet hierarchical indexing can be of two types, depending on whether there is a single metric per node or multiple set of metrics per node.  \n",
+    "\n",
+    "If a node contains a single metric, the DataFrame will use an `Index` object containing the node column. If a node has additional level of information, Hatchet creates a `MultiIndex` to store the information pertaining to multiple set of metrics per node. `MultiIndex` stores the node column as the \"top\" level of the index, followed by additional information on the levels below. \n",
```

Comment:
"If a node has additional level of information", should most likely change to "If a node has an additional level of information" or "If a node has additional levels of information". As well as "Multiple set of metrics" should most likely be changed to "Multiple sets of metrics".